subsys: power: remove device_pm_control_nop function

gmarull opened the pull request:
Devices that do not require PM should just use `NULL`. `device_pm_control_nop` is still kept as an alias to `NULL` untill all
in-tree usage is replaced with `NULL`. After that it can be either removed or deprecated to make out-of-tree users transition smoother.

Some devices have been ported (everything except `drivers/*`, the main user of `device_pm_control_nop`). I'll take care of porting remaining devices if this PR moves forward.

gmarull committed:
net: remove usage of device_pm_control_nop

If device PM is not implemented just use NULL.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
usb: remove usage of device_pm_control_nop

If device PM is not implemented just use NULL.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
arch: x86: remove usage of device_pm_control_nop

If device PM is not implemented just use NULL.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
soc: arm: st_stm32: remove usage of device_pm_control_nop

If device PM is not implemented just use NULL.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
power: remove device_pm_control_nop function

Devices that do not require PM should just use NULL.
`device_pm_control_nop` is still kept as an alias to NULL untill all
in-tree usage is replaced with NULL.

Code relying on device_pm_control function now returns -ENOTSUP
(equivalent to calling device_pm_control_nop).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

nashif commented on include/device.h:
```diff
@@ -781,15 +762,12 @@ static inline int device_set_power_state(const struct device *dev,
-						 DEVICE_PM_SET_POWER_STATE,
-						 &device_power_state, cb, arg);
+	if (dev->device_pm_control ==  NULL) {
+		return -ENOTSUP;
```
should return -ENOSYS (not implemented)

gmarull replied:
fixed, thanks (note: `-ENOTSUP` was returned from `device_pm_control_nop`).

nashif commented on include/device.h:
```diff
@@ -808,15 +786,12 @@ static inline int device_set_power_state(const struct device *dev,
-						 DEVICE_PM_GET_POWER_STATE,
-						 device_power_state, NULL, NULL);
+	if (dev->device_pm_control == NULL) {
+		return -ENOTSUP;
```
-ENOSYS

gmarull replied:
fixed

gmarull committed:
tests: kernel: device: adjust for device pm changes

- If device PM is not supported -ENOSYS is returned, update test case to
  account for that
- Remove usage of device_pm_control_nop

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>
